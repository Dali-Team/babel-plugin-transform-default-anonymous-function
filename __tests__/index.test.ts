import * as babel from "@babel/core";
import pluginTransformDefaultAnonymousFunction, {
  TRANSFORMED_DEFAULT_NAME,
} from "../src/index";

describe("babel-plugin-transform-default-anonymous-function", () => {
  it("should work with simple arrow function", () => {
    const actual = transform(`export default (a,b,c) => {};
`);
    expect(actual).toMatchSnapshot();
  });

  it("should work with arrow function with short form", () => {
    const actual = transform(`export default () => null;
`);
    expect(actual).toMatchSnapshot();
  });

  it("should work with simple anonymous function", () => {
    const actual = transform(`export default function() { return null };
`);
    expect(actual).toMatchSnapshot();
  });

  it("should work with high order component", () => {
    const actual = transform(`const Component = () => null;
export default memo(Component);
`);
    expect(actual).toMatchSnapshot();
  });

  it("should work with no conflict name", () => {
    const actual = transform(`const ${TRANSFORMED_DEFAULT_NAME} = '1';
export default () => {};
`);
    expect(actual).toMatchSnapshot();
  });
});

function transform(input: string, options: any = {}) {
  return babel.transform(input, {
    babelrc: false,
    configFile: false,
    envName: options.envName,
    presets: ["@babel/preset-react"],
    plugins: [
      [
        pluginTransformDefaultAnonymousFunction,
        {
          skipEnvCheck:
            options.skipEnvCheck === undefined ? true : options.skipEnvCheck,
          // To simplify debugging tests:
          ...options.freshOptions,
        },
      ],
      ...(options.plugins || []),
    ],
  })!.code!;
}
