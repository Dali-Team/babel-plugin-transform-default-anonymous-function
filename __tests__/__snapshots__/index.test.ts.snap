// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-transform-default-anonymous-function should work with arrow function with short form 1`] = `
"export default function Transformed_default_name_() {
  return null;
}"
`;

exports[`babel-plugin-transform-default-anonymous-function should work with high order component 1`] = `
"const Component = () => null;

const Transformed_default_name_ = memo(Component);
export default Transformed_default_name_;"
`;

exports[`babel-plugin-transform-default-anonymous-function should work with no conflict name 1`] = `
"const Transformed_default_name_ = '1';
export default function Transformed_default_name_Transformed_default_name__0() {}"
`;

exports[`babel-plugin-transform-default-anonymous-function should work with simple anonymous function 1`] = `
"export default function Transformed_default_name_() {
  return null;
}
;"
`;

exports[`babel-plugin-transform-default-anonymous-function should work with simple arrow function 1`] = `"export default function Transformed_default_name_(a, b, c) {}"`;
